### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by tobiasKaminsky**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", stable-[0-9][0-9]* ]
```
Comment (AlvaroBrey) [inline .github/workflows/codeql.yml:16]:
```suggestion
    branches: [ "master", "stable-[0-9][0-9]*" ]
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java', 'python', 'ruby' ]
```
Comment (AlvaroBrey) [inline .github/workflows/codeql.yml:35]: We should skip jdk install and gradle build for languages other than java. Can be done with a `if:` on setup and assemble steps.

**Review by AlvaroBrey — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (tobiasKaminsky) [reply .github/workflows/codeql.yml:35]: Do we need python / ruby at all?
I removed it in library…

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can we put this into the template repo so it can also be updated from a central place? :)

**Comment by AlvaroBrey** [comment]
> @tobiasKaminsky can we put this into the template repo so it can also be updated from a central place? :)

https://github.com/nextcloud/android-common/pull/1 <- need to create a github PAT (from nextcloud-android-bot for example) and add it as a secret to the common repo. I can't as I don't have admin permissions on that repo.

**Comment by AndyScherzinger** [comment]
> I can't as I don't have admin permissions on that repo.

@AlvaroBrey now you do 😃

**Pull request merged**